packages/zero-cache/src/services/replicator/replication-status.ts

```diff
@@ -0,0 +1,151 @@
+import type {LogContext} from '@rocicorp/logger';
+import {createSilentLogContext} from '../../../../shared/src/logging-test-utils.ts';
+import type {
+  ReplicatedIndex,
+  ReplicatedTable,
+  ReplicationStage,
+  ReplicationStatusEvent,
+  Status,
+} from '../../../../zero-events/src/status.ts';
+import type {Database} from '../../../../zqlite/src/db.ts';
+import {computeZqlSpecs, listIndexes} from '../../db/lite-tables.ts';
+import type {LiteTableSpec} from '../../db/specs.ts';
+import {
+  makeErrorDetails,
+  publishCriticalEvent,
+  publishEvent,
+} from '../../observability/events.ts';
+
+const byKeys = ([a]: [string, unknown], [b]: [string, unknown]) =>
+  a < b ? -1 : a > b ? 1 : 0;
+
+export class ReplicationStatusPublisher {
+  readonly #db: Database;
+  #timer: NodeJS.Timeout | undefined;
+
+  constructor(db: Database) {
+    this.#db = db;
+  }
+
+  publish(
+    lc: LogContext,
+    stage: ReplicationStage,
+    description?: string,
+    interval = 0,
+  ): this {
+    this.stop();
+    publishEvent(
```

Comment:
would it make more sense to make publishEvent a property that is passed in instead of overriding it for tests?